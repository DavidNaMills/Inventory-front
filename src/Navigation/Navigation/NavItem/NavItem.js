import React from 'react';
import classes from './NavStyle.module.scss';
import { NavLink } from 'react-router-dom';


const NavHome = ({ to, label, Icon=null, width='50px', isMobile=false, tempClose=null}) => {

    return (
        <NavLink 
            onClick={tempClose}
            to={to} 
            className={isMobile ? classes.mobileNavItem : classes.navItem} 
            style={{width}}
            activeClassName={classes.navItem__active}
        >
            {Icon && <Icon/>}
            {label}
        </NavLink>
    )
}

export default NavHome;