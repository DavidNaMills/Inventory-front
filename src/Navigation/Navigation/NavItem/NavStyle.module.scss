@import '../../../styles/variables/colors.module.scss';

.navItem{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    padding: 5px;
    color: $nav-color-inactive;
    // border-bottom: solid $nav-border 3px;
    
    &:hover{
        background-color: $nav-hover;
        color: $light-text;
    }

    &__active{
        background-color: $nav-active;
        color: $title-color;
    }
}

.mobileNavItem{
    display: flex;
    align-items: center;
    text-decoration: none;
    padding: 5px;
    color: $nav-color-inactive;
    
    &:hover{
        background-color: $nav-hover;
        color: $light-text;
    }

    &__active{
        background-color: $nav-active;
        color: $title-color;
    }
}