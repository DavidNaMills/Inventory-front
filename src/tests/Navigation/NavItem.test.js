import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter, NavLink } from 'react-router-dom';
import { mount } from 'enzyme';

import MenuBookIcon from '@material-ui/icons/MenuBook';
import NavItem from '../../Navigation/Navigation/NavItem/NavItem';


describe('<NavItem/> test suite', () => {
    it('matches snapshot', () => {
        const comp = renderer.create(<BrowserRouter>
            <NavItem to='/' label='test' />
        </BrowserRouter>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });

    it('exists', () => {
        const wrapper = mount(
            <BrowserRouter>
                <NavItem to='/' label='test' />
            </BrowserRouter>
        );
        expect(wrapper.exists).toBeTruthy();
    });

    it('renders with an icon', () => { 
        const wrapper = mount(
            <BrowserRouter>
                <NavItem to='/' label='test' Icon={MenuBookIcon}/>
            </BrowserRouter>
        );
        expect(wrapper.find(MenuBookIcon).length).toBe(1);
    });
});