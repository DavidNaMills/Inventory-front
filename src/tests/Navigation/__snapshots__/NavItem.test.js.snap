// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem/> test suite matches snapshot 1`] = `
<a
  aria-current="page"
  className="navItem navItem__active"
  href="/"
  onClick={[Function]}
  style={
    Object {
      "width": "50px",
    }
  }
>
  test
</a>
`;
